frontend/src/metabase/components/TextWidget/TextWidget.tsx

```diff
@@ -93,8 +93,11 @@ class TextWidget extends Component<Props, State> {
           changeFocus(true);
         }}
         onBlur={() => {
+          if (!this.state.value) {
+            setValue(null);
+          }
```

Comment:
@WiNloSt your idea looks solid to me

Reply:
I think the same issue happens with field filters (they use a different component). Fixing it in `setParameterValue` is not enough too since we use these components for default values etc. Not sure what would be the best path forward. I'd try fixing this issue with all parameter types and all places where they're used and see what fixes makes the most sense.

Reply:
I anticipate that fixing the issue in the affected components (like now) would be more straightforward in the end, but not sure.

Reply:
I checked the solution proposed by Kelvin and it works well and covers also number type, but field filters work differently indeed

Reply:
I decided to use normalization in redux as it covers 3 components at one place and with the same interface, but as for default value, it's a bit different, so I ended up with the "normalization" inside the component